tests/test_modeling_plbart.py

```diff
@@ -0,0 +1,655 @@
+    if a is None and b is None:
+        return True
+    try:
+        if torch.allclose(a, b, atol=atol):
+            return True
+        raise
+    except Exception:
+        pct_different = (torch.gt((a - b).abs(), atol)).float().mean().item()
+        if a.numel() > 100:
+            msg = f"tensor values are {pct_different:.1%} percent different."
+        else:
+            msg = f"{a} != {b}"
+        if prefix:
+            msg = prefix + ": " + msg
+        raise AssertionError(msg)
+
+
+def _long_tensor(tok_lst):
+    return torch.tensor(tok_lst, dtype=torch.long, device=torch_device)
+
+
+@require_torch
+@require_sentencepiece
+@require_tokenizers
+class AbstractSeq2SeqIntegrationTest(unittest.TestCase):
+    maxDiff = 1000  # longer string compare tracebacks
+    checkpoint_name = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.tokenizer = AutoTokenizer.from_pretrained(cls.checkpoint_name, use_fast=False)
+        return cls
+
+    @cached_property
+    def model(self):
+        """Only load the model if needed."""
+        model = PLBartForConditionalGeneration.from_pretrained(self.checkpoint_name).to(torch_device)
+        if "cuda" in torch_device:
+            model = model.half()
+        return model
+
+
+@require_torch
+@require_sentencepiece
+@require_tokenizers
+class PLBartJavaCsIntegrationTest(AbstractSeq2SeqIntegrationTest):
+    checkpoint_name = "uclanlp/plbart-java-cs"
+    src_text = [
+        "public int maximum(int a, int b, int c){return Math.max(a, Math.max(b, c));}",
+        "public int product(int a, int b, int c){return a*b*c;}",
+    ]
+    tgt_text = [
+        "public int maximum(int a, int b, int c){return Math.Max(",
+        "public int Product(int a, int b, int c){return a * b *",
+    ]
+
+    @slow
+    def test_java_cs_generate_one(self):
+        batch: BatchEncoding = self.tokenizer(
+            ["public int maximum(int a, int b, int c){return Math.max(a, Math.max(b, c));}"], return_tensors="pt"
+        ).to(torch_device)
+        translated_tokens = self.model.generate(**batch)
+        decoded = self.tokenizer.batch_decode(translated_tokens, skip_special_tokens=True)
+        self.assertEqual(self.tgt_text[0], decoded[0])
+        # self.assertEqual(self.tgt_text[1], decoded[1])
+
+    @slow
+    def test_java_cs_generate_batch(self):
+        batch: BatchEncoding = self.tokenizer(self.src_text, return_tensors="pt", padding=True, truncation=True).to(
+            torch_device
+        )
+        translated_tokens = self.model.generate(**batch)
+        decoded = self.tokenizer.batch_decode(translated_tokens, skip_special_tokens=True)
+        assert self.tgt_text == decoded
+
+    def test_plbart_java_cs_config(self):
+        plbart_models = ["uclanlp/plbart-java-cs"]
+        expected = {
+            "scale_embedding": True
+        }  # TODO: Check if "output_past": True is needed. Currently throws an error.
```

Comment:
What's the status of the TODO? It should be put on a separate line to be able to have the expected dict in one line.

Reply:
Currently, the `plbart-java-cs` model does not have `output_past` attribute in its configuration. Removing the `TODO`.